### PR TITLE
Messaging notifications

### PR DESCRIPTION
**Pull request by victor-paumier**
- improve flashbags
- Remove newsletter box from the registration page
- Add a notification part on the profile
- Create a command to send a mail to notify about unread message

### CODE REVIEW

File: assets/sass/components/_toggle-switch.scss [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  transform: translateX(26px);
+}
+
+.slider.round {
+  border-radius: 34px;
+}
+
+.slider.round:before {
+  border-radius: 50%;
+}
```
Comment (JeremieSamson) [inline assets/sass/components/_toggle-switch.scss]: line break

File: src/Controller/MessageController.php [hunk 2]
```diff
@@ -33,8 +33,18 @@ public function threadAction($threadId)
             $thread = $message->getThread();
             $this->get('fos_message.thread_reader')->markAsRead($thread);
 
+            $listMessages = $thread->getMessages();
+            $previousMessage = $listMessages[\count($listMessages) - 2];
+
+            if ($previousMessage->getCreatedAt()->format('Y-m-d') !== $message->getCreatedAt()->format('Y-m-d')) {
```
Comment (JeremieSamson) [inline src/Controller/MessageController.php]: what's about ternary condition ?

File: src/Controller/MotivationController.php [hunk 3]
```diff
@@ -210,7 +210,7 @@ public function delete(ObjectManager $om, Request $request, Motivation $motivati
             $om->remove($motivation);
             $om->flush();
 
-            $this->addFlash('check', 'Motivation Deleted');
+            $this->addFlash('success', 'Motivation Deleted');
```
Comment (JeremieSamson) [inline src/Controller/MotivationController.php]: IMHO even admin should have translation

File: src/Controller/SectionController.php [hunk 4]
```diff
@@ -48,11 +48,11 @@ public function new(ObjectManager $om, Request $request)
                 $om->persist($section);
                 $om->flush();
 
-                $this->addFlash('check', 'Section saved');
+                $this->addFlash('success', 'Section saved');
```
Comment (JeremieSamson) [inline src/Controller/SectionController.php]: same here

File: src/Controller/SectionController.php [hunk 5]
```diff
@@ -48,11 +48,11 @@ public function new(ObjectManager $om, Request $request)
...
 
                 return $this->redirectToRoute('section_index');
             }
-            $this->addFlash('times', 'Their is some errors in the section');
+            $this->addFlash('error', 'Their is some errors in the section');
```
Comment (JeremieSamson) [inline src/Controller/SectionController.php]: same here

File: src/Repository/UserRepository.php [hunk 6]
```diff
@@ -175,6 +175,25 @@ public function getUsersSearchAutocomplete($search, $limit = null, $offset = nul
         return $qb->getQuery()->getResult();
     }
 
+    public function getUsersWithUnreadMessageToNotify(int $hoursInterval = 2)
```
Comment (JeremieSamson) [inline src/Repository/UserRepository.php]: typehint

File: src/Repository/UserRepository.php [hunk 7]
```diff
@@ -175,6 +175,25 @@ public function getUsersSearchAutocomplete($search, $limit = null, $offset = nul
...
+    {
+        return $this->createQueryBuilder('u')
+            ->distinct()
+            ->innerJoin('u.messageMetadatas', 'mm')
+            ->innerJoin('mm.message', 'm')
+            ->innerJoin('u.userNotification', 'notif')
+            ->where('notif.newMessage = :newMessage')
+            ->andWhere('mm.isRead = :isRead')
```
Comment (JeremieSamson) [inline src/Repository/UserRepository.php]: Read red red, `isRead` should be `isRed`

Comment (victor-paumier) [reply src/Repository/UserRepository.php]: Read read read...

File: src/Controller/MotivationController.php (hunk 3 above)
Comment (victor-paumier) [reply src/Controller/MotivationController.php]: That's a type, not a translation. See Flashbag.js

File: src/Repository/UserRepository.php (hunk 6 above)
Comment (victor-paumier) [reply src/Repository/UserRepository.php]: which type?

Comment (JeremieSamson) [reply src/Repository/UserRepository.php]: function

File: assets/sass/components/_toggle-switch.scss (hunk 1 above)
Comment (victor-paumier) [reply assets/sass/components/_toggle-switch.scss]: done

File: src/Controller/MessageController.php (hunk 2 above)
Comment (victor-paumier) [reply src/Controller/MessageController.php]: done

File: src/Repository/UserRepository.php (hunk 6 above)
Comment (victor-paumier) [reply src/Repository/UserRepository.php]: done

**Pull request merged**